### DISCUSSION TITLE
Build challenge with active-artemis-operator

### DISCUSSION THREAD

**Discussion by funkyjive** [arkmq-org/arkmq-org-broker-operator#424] [General] [answered]
I am following the instructions on the quick start to try this out.  I am attempting to build the operator 
(https://artemiscloud.io/docs/help/building/)

but as I attempt to do the make build, I get this error: 

```
myuser@svr:~/dev/activemq-artemis-operator$ make build
go: creating new go.mod: module tmp
Downloading sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0
go: added github.com/fatih/color v1.12.0
go: added github.com/go-logr/logr v0.4.0
go: added github.com/gobuffalo/flect v0.2.3
go: added github.com/gogo/protobuf v1.3.2
go: added github.com/google/go-cmp v0.5.6
go: added github.com/google/gofuzz v1.1.0
go: added github.com/inconshreveable/mousetrap v1.0.0
go: added github.com/json-iterator/go v1.1.11
go: added github.com/mattn/go-colorable v0.1.8
go: added github.com/mattn/go-isatty v0.0.12
go: added github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: added github.com/modern-go/reflect2 v1.0.1
go: added github.com/spf13/cobra v1.2.1
go: added github.com/spf13/pflag v1.0.5
go: added golang.org/x/mod v0.4.2
go: added golang.org/x/net v0.0.0-20210520170846-37e1c6afe023
go: added golang.org/x/sys v0.0.0-20210616094352-59db8d763f22
go: added golang.org/x/text v0.3.6
go: added golang.org/x/tools v0.1.5
go: added golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: added gopkg.in/inf.v0 v0.9.1
go: added gopkg.in/yaml.v2 v2.4.0
go: added gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: added k8s.io/api v0.22.2
go: added k8s.io/apiextensions-apiserver v0.22.2
go: added k8s.io/apimachinery v0.22.2
go: added k8s.io/klog/v2 v2.9.0
go: added k8s.io/utils v0.0.0-20210819203725-bdf08cb9a70a
go: added sigs.k8s.io/controller-tools v0.7.0
go: added sigs.k8s.io/structured-merge-diff/v4 v4.1.2
go: added sigs.k8s.io/yaml v1.2.0
/home/myuser/dev/activemq-artemis-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
bash: line 1: /home/myuser/dev/activemq-artemis-operator/bin/controller-gen: No such file or directory
make: *** [Makefile:111: generate] Error 127

```

I have no experience with go, but I was digging around the makefile and trying to figure it out and it appears that the go-get-tool inside the makefile is meant to put a file in activemq-artemis-operator/bin, but that is clearly not happening.  I am not sure how go is supposed to work for this.

I was successful attempting to do those instructions manually by manually getting the latest code from:

`https://github.com/kubernetes-sigs/controller-tools/tree/master/cmd/controller-gen`

and doing: 

`go build -o ~/dev/activemq-artemis-operator/bin/controller-gen`

and I was able to get by this, but then immediately ran into problems with kustomize.

I am not sure what I am doing wrong, other than I wondered if my problem is that I my go was the latest 1.19?

`go version go1.19.4 linux/amd64`

The go installation took me to the latest 1.19 instead of 1.17.   I can attempt to downgrade to 1.17 if that will really solve it.

**Comment by funkyjive** [comment, marked as answer]
Ok, so I changed to 1.17 and the build worked.  I'll follow directions better next time.

**Reply by brusdev** [reply]
@funkyjive thansk for your feedback
